kernel/packages/decentraland-loader/store/download/downloadState.ts

```diff
@@ -0,0 +1,58 @@
+import { StringCoordinate, SceneContentServerEntity } from '../sceneInfo/sceneInfo'
+import { mapAdd } from '../utils/mapAdd'
+import { mapRemove } from '../utils/mapRemove'
+import { unique } from '../utils/unique'
+import { setFilter } from '../utils/setFilter'
+import { setRemove } from '../utils/setRemove'
+
+export type DownloadState = {
+  queued: StringCoordinate[]
+  pendingDownloads: Record<StringCoordinate, boolean>
+  emptyValues: Record<StringCoordinate, boolean>
```

Comment:
Why not use an array both structures?

Reply:
I was looking into making it more efficient for arbitrary add/deletes (simulating a Set). Now that you mention it, I'd like to review this. I'm also thinking about dropping the `empty`